File: htm-css-js/02/dart/main.dart
import 'dart:html';

void main() {
  // Browser info
  String txtBrow = 'Browser info: ';
  txtBrow += window.navigator.appCodeName;
  txtBrow += ' - ';
  txtBrow += window.navigator.appVersion;
  querySelector('#browser-info').innerHtml = txtBrow;
  // Screen info
  String txtScreen = 'Screen info: ';
  txtScreen += window.screen.width.toString();
  txtScreen += 'x';
  txtScreen += window.screen.height.toString();
  querySelector('#screen-info').innerHtml = txtScreen;

  // New window
  WindowBase opened; 
  querySelector('#click1').onClick.listen((e) => opened = newWindow());
  // Page prev (3rd)
  querySelector('#histo-prev').onClick.listen((e) => doPagePrev());
  // Page next (1st)
  querySelector('#histo-next').onClick.listen((e) => doPageNext());


  // close new window
  querySelector('#click2').onClick.listen((e) => closeWindow(opened));
}

WindowBase newWindow() => window.open("infos.html", "String target,", "_blank");


void doPagePrev () {
  window.history.back();
  window.history.go(-3);
}
void doPageNext () {
  window.history.forward();
}

closeWindow(WindowBase wb) => wb.close();

File: htm-css-js/02/js/main.js
var btn1 = document.getElementById('click1');
var btn2 = document.getElementById('click2');
var histoP = document.getElementById('histo-prev');
var histoN = document.getElementById('histo-next');

btn1.addEventListener('click', (e) => doOnClickBtn1(e));
btn2.addEventListener('click', (e) => doOnClickBtn2(e));
histoP.addEventListener('click', (e) => doHistoPrev(e));
histoN.addEventListener('click', (e) => doHistoNext(e));


function doOnClickBtn1 (e) {
	window.open("infos.html", "Informations", "width=500px,height=500px");
}
function doOnClickBtn2 (e) {}

var txt = 'Browser:' + navigator.appCodeName + ' - ' + navigator.appVersion;
document.getElementById('browser-info').innerHTML = txt;


function doHistoPrev(e){
	// window.history.back();
	window.history.go(-3);
}
function doHistoNext(e){
	window.history.forward();
	
}

txt = 'Screen:' + window.screen.width + 'x' + window.screen.height;
document.getElementById('screen-info').innerHTML = txt;